workloads/oslat/common.sh

```diff
@@ -0,0 +1,272 @@
+      log "Trying to find a suitable node for oslat"
+      # TODO: check if there are two -rt nodes already and use one of them
+      # iterate over worker nodes bareMetalHandles until we have 2
+      worker_count=0
+      oslat_workers=()
+      workers=$(oc get nodes | grep ^worker | awk '{print $1}')
+      while [ $worker_count -lt 1 ]; do
+        for worker in $workers; do
+       	  worker_ip=$(oc get node $worker -o json | jq -r ".status.addresses[0].address" | grep 192 )
+          if [[ ! -z "$worker_ip" ]]; then
+            oslat_workers+=( $worker )
+            ((worker_count=worker_count+1))
+	    break
+          fi
+        done
+      done
+    fi
+  fi
+  # label the two nodes for the performance profile
+  log "Labeling -lat nodes"
+  for w in ${oslat_workers[@]}; do
+    oc label node $w node-role.kubernetes.io/worker-rt="" --overwrite=true
+  done
+  # create the machineconfigpool
+  log "Creating the MCP"
+  oc apply -f machineconfigpool.yaml
+  sleep 30
+  if [ $? -ne 0 ] ; then
+    log "Couldn't create the MCP, exiting!"
+    exit 1
+  fi
+  # add the label to the MCP pool 
+  log "Labeling the MCP"
+  oc label mcp worker-rt machineconfiguration.openshift.io/role=worker-rt --overwrite=true
+  if [ $? -ne 0 ] ; then
+    log "Couldn't label the MCP, exiting!"
+    exit 1
+  fi
+  # apply the performanceProfile
+  log "Applying the performanceProfile since it doesn't exist yet"
+  profile=$(oc get performanceprofile benchmark-performance-profile-0 --no-headers)
+  if [ $? -ne 0 ] ; then
+    log "PerformanceProfile not found, creating it"
+    oc apply -f perf_profile.yaml
+    if [ $? -ne 0 ] ; then
+      log "Couldn't apply the performance profile, exiting!"
+      exit 1
+    fi
+  fi
+  # We need to wait for the nodes with the perfProfile applied to to reboot
+  # this is a catchall approach, we sleep for 60 seconds and check the status of the nodes
+  # if they're ready we'll continue. Should the performance profile require reboots, that will have
+  # started within the 60 seconds
+  log "Sleeping for 60 seconds"
+  sleep 60
+  readycount=$(oc get mcp worker-rt --no-headers | awk '{print $7}')
+  while [[ $readycount -lt 1 ]]; do
+    log "Waiting for -rt nodes to become ready again, sleeping 1 minute"
+    sleep 60
+    readycount=$(oc get mcp worker-rt --no-headers | awk '{print $7}')
+  done
+}
+
+deploy_operator() {
+  deploy_benchmark_operator ${OPERATOR_REPO} ${OPERATOR_BRANCH}
+  if [[ $? != 0 ]]; then
+     exit 1
+  fi
+  rm -rf benchmark-operator
+}
+
+run_workload() {
+  log "Deploying oslat benchmark"
+  local TMPCR=$(mktemp)
+  envsubst < $1 > ${TMPCR}
+  run_benchmark ${TMPCR} ${TEST_TIMEOUT}
+  local rc=$?
+  if [[ ${TEST_CLEANUP} == "true" ]]; then 
+    log "Cleaning up benchmark"
+    kubectl delete -f ${TMPCR}
+  fi
+  return ${rc}
+}
+
+check_logs_for_errors() {
+client_pod=$(oc get pods -n benchmark-operator --no-headers | awk '{print $1}' | grep oslat | awk 'NR==1{print $1}')
```

Comment:
If there are previous run this could catch a previously completed pod. Might want to ensure its running.